driver/arguments.py

```diff
@@ -84,7 +84,7 @@
 Examples:
 
 %s
-""" % "\n\n".join("%s\n%s" % (desc, " ".join(cmd)) for desc, cmd in EXAMPLES)
+""" % "\n\n".join("%s\n%s" % (desc, " ".join(["./fast-downward.py"] + parameters)) for desc, parameters in EXAMPLES)
```

Comment:
add sys.executable prior to ./fast-downward.py?

Reply:
Nope, this is just print to command line when running `fast-downward.py --help`.

Reply:
Was there something broken about the old version? "./" makes the assumption that people are calling this from the current directory, which doesn't have to be true. One could use sys.args[0] to find out how they actually called it, but the actual unix convention is to:

1) compute the filename programmatically rather than hard-coding it (i.e., use sys.args[0]), but
2) strip the path

The advantage of 1) is that it adapts if people rename or symlink the file. The motivation for 2) is unclear to me, but nevertheless it's the expected behaviour.

So according to this convention, we should use `os.path.basename(sys.args[0])`.

Reply:
PS: I see now that the old version was no better and the new version is better than the old one. (github didn't give enough context for this, but I've now looked at the full pull request.) Still, I suggest if we're improving this that we also make the change from my previous message.

Reply:
Using `os.path.basename(sys.args[0])` makes `./fast-downward.py --help` called from the same directory print lines like `fast-downward.py --validate ....` Is this what you want? I thought the original purpose was to allow people to directly copy the example line, which is now not possible anymore, neither on linux (where it worked before) nor on windows (where you would use `python fast-downward.py` or maybe other syntax I don't know).

Reply:
P.S.: the reason I had to change this is because the EXAMPLE variable is used from the driver tests, and there I needed to change all calls from `./fast-downward.py` to `sys.executable fast-downward.py`. Do you think we should also determine the script name automatically there?

Reply:
I think it's not necessary to follow all such "best practices" in the tests. In general, please go with whatever you prefer in the tests. BTW, I did a little test with "grep" (as a typical Unix tool) to verify that I remembered the two things (use filename programmatically, strip directory name) correctly, and indeed grep seems to follow this:

```
$ grep -u
Usage: grep [OPTION]... PATTERN [FILE]...
Try 'grep --help' for more information.
$ cp /bin/grep foo/mygrep
$ ln -s /bin/grep myothergrep
$ foo/mygrep -u
Usage: mygrep [OPTION]... PATTERN [FILE]...
Try 'mygrep --help' for more information.
$ ./myothergrep -u
Usage: myothergrep [OPTION]... PATTERN [FILE]...
Try 'myothergrep --help' for more information.
```